File: frontend/src/app/core/models/watches/watch-filter-request.dto.ts
import { WatchCondition } from './enums/watch-condition';
import { WatchGender } from './enums/watch-gender';
import { WatchMovementType } from './enums/watch-movement-type';
import { WatchStatus } from './enums/watch-status';
import { WatchType } from './enums/watch-type';

export interface WatchFilterRequestDTO {
    manufacturer?: string | null;
    model?: string | null;
    referenceNumber?: string | null;
    serialNumber?: string | null;
    movement?: string | null;
    condition?: WatchCondition | null;
    gender?: WatchGender | null;
    movementType?: WatchMovementType | null;
    status?: WatchStatus | null;
    watchType?: WatchType | null;
    branch?: string | null;
    minPrice?: number | null;
    maxPrice?: number | null;
    minYear?: number | null;
    maxYear?: number | null;
}

File: backend/src/main/java/com/yourwatchrental/watchrental/watch/dto/request/WatchFilterRequestDTO.java
package com.yourwatchrental.watchrental.watch.dto.request;

import com.yourwatchrental.watchrental.watch.enums.*;
import jakarta.validation.constraints.PositiveOrZero;

import java.math.BigDecimal;
import java.util.UUID;

public record WatchFilterRequestDTO(
        UUID watchId,
        String manufacturer,
        String model,
        String referenceNumber,
        String serialNumber,
        String movement,
        Condition condition,
        Gender gender,
        MovementType movementType,
        Status status,
        WatchType watchType,
        UUID branchId,

        @PositiveOrZero(message = "Minimum price needs to be accurate")
        BigDecimal minPrice,
        @PositiveOrZero(message = "Maximum price needs to be accurate")
        BigDecimal maxPrice,


        @PositiveOrZero(message = "Minimum year needs to be accurate")
        Integer minYear,
        @PositiveOrZero(message = "Maximum year needs to be accurate")
        Integer maxYear
) {
}
